### PR TITLE
Update show_cts.py

### LINKED ISSUE #521: IOSXE - show cts role-based permissions fails parsing
The following cli output gives an error when trying to parse it:
```
m-bcp-c9516-01#show cts role-based permissions
IPv4 Role-based permissions from group Invalid to group Invalid:
Permit IP-00
RBACL Monitor All for Dynamic Policies : FALSE
RBACL Monitor All for Configured Policies : FALSE
```

image

looks like in file src/genie/libs/parser/iosxe/show_cts.py
lines 1933, 1944 tries to access nonexistent collection key cts_rb_permissions_dict['indexes']

### PR DESCRIPTION
**Pull request by komurzak-cisco**
updated 'show cts role-based permissions' parser to work on actual output

## Description
This PR fixes  #521 a bug when code is trying to access uninitialized 'indexes' key of a dictionary.

## Motivation and Context
Show cts role-based permissions parses was giving error when trying to parse this cli output:
```
IPv4 Role-based permissions from group Invalid to group Invalid:
        Permit IP-00
RBACL Monitor All for Dynamic Policies : FALSE
RBACL Monitor All for Configured Policies : FALSE

```
This PR will fix parsing output of show cts role-based permissions when there are no policies present.
## Impact (If any)
No negative impact

## Screenshots:
image

## Checklist:

- [x] I have updated the changelog.
- [x] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [ ] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

File: changelog/2021/changelog_komurzak_iosxe_show_cts_20210830.rst [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+                            Fix
+--------------------------------------------------------------------------------
+* IOSXE
+    * Modified ShowCtsRoleBasedPermissionsSchema:
+        * Modified int to be Optional inside 'indexes' key
+    * Modified ShowCtsRoleBasedPermissions:
+        * Initialized 'indexes' key by default so that there is no runtime error if collection is empty
+
+
+
```
Comment (domachad) [inline changelog/2021/changelog_komurzak_iosxe_show_cts_20210830.rst:11]: Please remove blank lines

**Review by domachad — CHANGES_REQUESTED** [review]
Thank you four your contribution, @komurzak-cisco!
Can you please add a unit test that supports this change?

File: changelog/2021/changelog_komurzak_iosxe_show_cts_20210830.rst (hunk 1 above)
Comment (komurzak-cisco) [reply changelog/2021/changelog_komurzak_iosxe_show_cts_20210830.rst]: removed

**Comment by komurzak-cisco** [comment]
@domachad, unit test added, blank lines removed

File: src/genie/libs/parser/iosxe/show_cts.py [hunk 2]
```diff
@@ -1802,7 +1802,7 @@ class ShowCtsRoleBasedPermissionsSchema(MetaParser):
 
     schema = {
         "indexes": {
-            int: {
+            Optional(int): {
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/show_cts.py:1805]: what kind of output will we see this as Optional?

File: src/genie/libs/parser/iosxe/tests/ShowCtsRoleBasedPermissions/cli/equal/sample_output_expected.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+expected_output = {
+    "indexes": {
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/ShowCtsRoleBasedPermissions/cli/equal/sample_output_expected.py:2]: based on the change, below can be possible, right?
```
expected_output = {
    "indexes": {}
}
```
could you please add the output as unittest?

File: src/genie/libs/parser/iosxe/tests/ShowCtsRoleBasedPermissions/cli/equal/sample_output_expected.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+        1: {
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/ShowCtsRoleBasedPermissions/cli/equal/sample_output_expected.py:3]: maybe some keyword from below and have key name for this info?
```
IPv4 Role-based permissions from group Invalid to group Invalid:
```

File: src/genie/libs/parser/iosxe/tests/ShowCtsRoleBasedPermissions/cli/equal/sample_output_expected.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+            "action_policy": "Permit",
+            "action_policy_group": "IP-00"
+        },
+        "monitor_dynamic": False,
+        "monitor_configured": False
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/ShowCtsRoleBasedPermissions/cli/equal/sample_output_expected.py:8]: even from parsed output, we could guess what they are. but I assume this name is very short. and doesn't tell this is for RBACL related. please consider updating schema with better key name.

**Review by tahigash — CHANGES_REQUESTED** [review]
reviewed and left comments

**Pull request closed without merging**